File: JavaScript/Basics/0.5-While-Loops/Exercises/walking.js
function walking(input) {
    let i = 0; let goal = 10000; let steps = 0;
    while (steps < goal || i < input.length) {
        let x = input[i];
        if (x == 'Going home') {
            steps += Number(input[i + 1]);
            if (steps < goal) {
                return `${Math.abs(steps - goal)} more steps to reach goal.`;
            }
            break;
        }
        steps += Number(x); i++;
    }
    return `Goal reached! Good job!\n${Math.abs(steps - goal)} steps over the goal!`;
}


console.log(walking(["1000",
"1500",
"2000",
"6500"]));
console.log(walking(["1500",
"3000",
"250",
"1548",
"2000",
"Going home",
"2000"]));
console.log(walking(["1500",
"300",
"2500",
"3000",
"Going home",
"200"]));
console.log(walking(["125",
"250",
"4000",
"30",
"2678",
"4682"]));

File: Python/0.1-Basics/0.4-While-Loops/Exercise/walking.py
steps_goal = 10000
steps_counter = 0

while steps_counter < steps_goal:
    current_steps = input()
    if current_steps == "Going home":
        current_steps = int(input())
        steps_counter += current_steps
        break

    current_steps = int(current_steps)
    steps_counter += current_steps

steps_diff = abs(steps_goal - steps_counter)

if steps_counter >= steps_goal:
    print(f"Goal reached! Good job!\n"
          f"{steps_diff} steps over the goal!")
else:
    print(f"{steps_diff} more steps to reach goal.")
